Dockerfile

```diff
@@ -6,7 +6,9 @@ COPY package*.json ./
 
 RUN npm install
 
-FROM node:14-alpine
+FROM node:14-slim
+
+RUN apt-get update && apt-get install -y secure-delete && apt-get clean
```

Comment:
I couldn't find any documentation about where `secure-delete` is actually used in this project. Is it run manually as part of maintenance?

A comment explaining why we're pulling in secure-delete would be handy.

Reply:
I'm not entirely sure either. It was bundled into the [original dockerfile](https://github.com/matrix-org/matrix-content-scanner/blob/v1.6.3/Dockerfile) which I then replaced with an alpine version [forgetting to remove it from the example config](https://github.com/matrix-org/matrix-content-scanner/blob/main/config/docker.config.yaml#L28). Later I realized the alpine version is hard to build on for a case we have in EMS, so would suggest reverting back to the debian base (though -slim), keeping secure-delete bundled in as it seems to work.

I think the original dockerfile was never meant for production and I can only assume that `secure-delete` was pulled in to either test for some case or as a known good tool to more securely delete the decrypted files.

I'm happy to drop the `secure-delete` from the dockerfile and anyone wanting to use it can just install it in their own docker file on top of the base (for example us in EMS). I'm having a hard time justifying it being here as it's tricky to find information why it's better, other than promises in the source repo. The last version is from 2013 (https://metadata.ftp-master.debian.org/changelogs//main/s/secure-delete/secure-delete_3.1-6_changelog) so not massively convincing.

Reply:
While it might not be the best tool to use, I think it's likely better than nodeJS `unlink` at least. `unlink` apparently calls the `unlink` command, which doesn't appear to actually attempt to zero out the contents of the space on disk: https://man7.org/linux/man-pages/man2/unlink.2.html

Tchap have requested we keep this in as `srm` is their preferred choice, and I don't think it's a bad idea to have it be the default in an environment where we can control what can be installed natively.

I would like to have a comment just explaining that the purpose of installing the package is to provide a more secure file erasure mechanism, and that using the `srm` command is specified in the [default docker config file](https://github.com/matrix-org/matrix-content-scanner/blob/main/config/docker.config.yaml#L28).

Reply:
Agree. Notes have been added :+1: